components/dashboard/src/projects/NewProject.tsx

```diff
@@ -500,13 +500,14 @@ function GitProviders(props: {
         setErrorMessage(undefined);
 
         const token = await getGitpodService().server.getToken({ host: ap.host });
-        if (token) {
+        const isGitHubEnterprise = ap.authProviderType === "GitHub" && ap.host !== "github.com";
```

Comment:
Not introduced on this PR, but we should have a narrower type here than a `string`, ie a union like `"GitHub" | "Gitlab" | ... `.

Reply:
Good point @andrew-farries! I agree with you, but since this implies modifying a central protocol symbol, I'll not address it here, but leave it for this follow-up issue: https://github.com/gitpod-io/gitpod/issues/8788